hikari/impl/event_manager_base.py

```diff
@@ -235,7 +239,13 @@ def dispatch(self, event: event_manager.EventT_inv) -> asyncio.Future[typing.Any
                     else:
                         future.set_result(event)
 
-                    waiter_set.remove((predicate, future))
+                except asyncio.InvalidStateError:
+                    pass
```

Comment:
As I said in the other comment, we shouldn't be needing to handle this error, as we should never encounter it if we remove the future after it is completed.

Reply:
Welp if you fine with event dispatch being stopped by cancelled futures then you do you ig 🤷‍♀️

Reply:
The bug is still there, I just dont think that this is the way to go about fixing it. This will just suppress it, not actually fix it. Unless there is another case when a future will be cancelled that is out of our control (which is then causing this issue), I don't think this is the way to fix it

Reply:
Well without that try except other listeners may miss an event

Reply:
Turns out I'm just being dumb and this is the best way to do it. Might also be best to check `if future.done()` and not even run the predicate, just remove the waiter from the mapping instead of catching the error.